### ISSUE TITLE
[DATA] Data inconsistency (different values for same metric) MU period:2013-02-28 10-Q

### ISSUE THREAD

**Issue by wei-jianlin** [dgunning/edgartools#814]
## Data Quality Issue Details
Data inconsistency (different values for same metric)

Environment                                                                                                                                
     EdgarTools Version: 5.31.2                                                                         
     Python Version: 3.12                                                                               
     Operating System: Linux                                                                            

     Bug Description                                                                                    
     What you expected to happen:                                                                      
     company.financials.get_net_income() 应该返回 MU (Micron Technology) 2013年Q2 (period: 2013-02-                                             
     28) 的净亏损 -$284,000,000                                                                                                                 
 
     What actually happened:                                                                                                                    
     get_net_income() 返回了 2,000,000，这是一个完全错误的正值
	 
 Error message (if any):                                                                           
                                                                                                           
     Revenue: 2078000000.0                                                                               
     Net Income: 2000000.0  # 错误！应该是 -284000000                                                   
                                                                                                       
     Reproduction                                                                                        
     Company/Ticker: MU (Micron Technology)                                                                                                     
     Time Period: Q2 2013 (period: 2013-02-28)                                                                                                  
     Relevant Forms: 10-Q 	 
	 
 Minimal code to reproduce:                                                                                              
                                                                                                                                                
     from edgar import set_identity, Company                                                              
     set_identity('your-email@example.com')                                                              
                                                                                                       
     company = Company('MU')                                                                            
     filings = list(company.get_filings(form='10-Q', amendments=False))                                     
                                                                                                           
     for f in filings:                                                                                                                          
         if '2013' in str(f.filing_date):                                                                                                       
             obj = f.obj()                                                                                                                      
             if str(obj.period_of_report) == '2013-02-28':                                                                                      
                 fin = obj.financials                                                                                                           
                 print('Revenue:', fin.get_revenue())
				 print('Net Income:', fin.get_net_income())  # 返回 2000000，错误！                                 
                 # HTML文件显示正确值: Net loss = -$284,000,000                                         
                                                                                                        
     Additional Context                                                                                  
     SEC 官网原始 10-Q 文件 (https://www.sec.gov/Archives/edgar/data/723125/0000723125-13-000042-          
     index.html) 显示：                                                                                 
     - Net sales (Revenue): $2,078 million ✓                                                                                                    
     - Net loss: $(284) million ✗ (edgar库返回 +2 million)                                                                                      
                                                                                                                                                
     这是一个数据质量问题，XBRL解析可能读取了错误的context或concept。

**Comment by wei-jianlin** [comment]
不是-$284000000，是-$286000000

**Comment by wei-jianlin** [comment]
[financial_data_202605171449.json](https://github.com/user-attachments/files/27892755/financial_data_202605171449.json)
The data in this file shows a net income (net_income) of 0, but the value obtained using XBRL is not 0. The net income is obtained using the get_net_income() method, and report_type equals 1 for 10-Q and 0 for 10-K. Please fix this issue, thank you

**Comment by dgunning** [comment]
Thanks for the detailed report — reproduction was clean on my end. Confirming this is a **P0 data-accuracy bug** and we'll ship a fix in the next point release.

## Reproduction (confirmed on 5.31.2)

```
Revenue:    2,078,000,000   ← correct
Net Income:     2,000,000   ← WRONG (should be -286,000,000)
```

## Root cause

`Financials.get_net_income()` matches by **label regex** against income-statement rows instead of by XBRL concept. For Micron's Q2 2013 10-Q the relevant rows are:

| # | Label | Concept | Value |
|---|---|---|---|
| 31 | "Net income attributable to noncontrolling interests" | `us-gaap:NetIncomeLossAttributableToNoncontrollingInterest` | +$2M |
| 32 | "Net **loss** attributable to Micron" | `us-gaap:NetIncomeLoss` | -$286M ← correct |

The regex `^Net Income` matches row 31 ("Net **i**ncome...") and returns the noncontrolling-interest line. Row 32's label starts with "Net **loss**" so none of the label patterns ever reach it. (Your original $284M vs $286M correction is just the dimensional breakdown — `us-gaap:ProfitLoss` is -$284M, `us-gaap:NetIncomeLoss` attributable to Micron is -$286M. -$286M is the canonical "net income" answer.)

This affects any filer who (a) reports a net loss in their label and (b) has a separate noncontrolling-interest row — likely many filings silently.

## Fix shape

Mirror `get_operating_income()`, which already does the right thing: concept-based lookup via the standardization mapping first, with `NetIncomeLossAttributableToNoncontrollingInterest` excluded. Label patterns become last-resort fallback.

## Next steps

- Tracked internally as `edgartools-sucu` (P0)
- Targeting point release **v5.31.3** alongside #812
- Regression test in `tests/issues/regression/test_issue_814.py` will assert MU Q2 2013 net income == -286,000,000
- Verification matrix across 5+ companies (NCI + non-NCI, gain + loss) to guard against re-regression

About the JSON file you attached — `net_income = 0` there is a separate but related symptom (the standardization layer falling through to a default rather than raising). The XBRL `NetIncomeLoss` value is present in the filing, so the fix above should resolve it. If you still see 0 after the patch, please re-attach with a fresh repro and we'll dig in.

Will update here when the patch lands.

**Comment by dgunning** [comment]
Fixed in [v5.31.3](https://github.com/dgunning/edgartools/releases/tag/v5.31.3). The fix has three parts: exact-match concept lookup in the standardization helper (resolves the NCI-row collision), concept-first `get_net_income()` that mirrors `get_operating_income()`, and IFRS `ifrs-full_ProfitLoss` added to the `Profit or Loss` concept mapping (fixes 20-F filers whose row label isn't "Net income"). Thanks for the clear repro — verification matrix now locks in 10 filers across US GAAP and IFRS.

**Comment by wei-jianlin** [comment]
thank you
